### PR TITLE
Update contour to 1.3

### PR DESCRIPTION
**Pull request by shashwathi**
summary: Update contour to 1.3

description: Created a patch file in hack directory. Updated update-deps file to apply the patch. Serving ingress conformance tests pass for this version of contour.

- Questions
- Is there a better way to keep the patch file in repo?
- Are there any suggestions on another approach to create and store the patch file?

TODO
- Updating docs. I will do this when everybody has agreed on an approach
 
/assign @mattmoor

### CODE REVIEW

File: hack/contour.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
+diff --git a/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go b/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go
+index 7f49df2..d2797e0 100644
+--- a/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go
++++ b/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go
+@@ -28,12 +28,18 @@ var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1"}
+ var HTTPProxyGVR = SchemeGroupVersion.WithResource("httpproxies")
+ var TLSCertificateDelegationGVR = SchemeGroupVersion.WithResource("tlscertificatedelegations")
+
```
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
++var SchemeBuilder = runtime.NewSchemeBuilder(AddKnownTypes)
++
++// AddToScheme applies the SchemeBuilder functions to a specified scheme
++var AddToScheme = SchemeBuilder.AddToScheme
++
+ // Resource gets an Contour GroupResource for a specified resource
+ func Resource(resource string) schema.GroupResource {
+ 	return SchemeGroupVersion.WithResource(resource).GroupResource()
+ }
+
```
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch (hunk 1 above)
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch (hunk 2 above)
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch [hunk 3]
```diff
@@ -0,0 +1,30 @@
+diff --git a/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go b/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go
+index d2797e0..7f49df2 100644
+--- a/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go
++++ b/vendor/github.com/projectcontour/contour/apis/projectcontour/v1/register.go
+@@ -28,18 +28,12 @@ var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1"}
+ var HTTPProxyGVR = SchemeGroupVersion.WithResource("httpproxies")
+ var TLSCertificateDelegationGVR = SchemeGroupVersion.WithResource("tlscertificatedelegations")
+
```
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+-var SchemeBuilder = runtime.NewSchemeBuilder(AddKnownTypes)
+-
+-// AddToScheme applies the SchemeBuilder functions to a specified scheme
+-var AddToScheme = SchemeBuilder.AddToScheme
+-
+ // Resource gets an Contour GroupResource for a specified resource
+ func Resource(resource string) schema.GroupResource {
+ 	return SchemeGroupVersion.WithResource(resource).GroupResource()
+ }
+
```
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch (hunk 1 above)
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch (hunk 2 above)
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch (hunk 1 above)
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

File: hack/contour.patch (hunk 2 above)
Comment (mattmoor) [inline hack/contour.patch]: Remove trailing whitespace:
```suggestion

```

**Comment by andrew-su** [comment]
@googlebot I consent.

**Comment by mattmoor** [comment]
/lgtm
/approve

Sweet!

**Pull request merged**